cli/config/configurator.go

```diff
@@ -106,6 +116,11 @@ func (c Configurator) Start(ctx context.Context, prev *Config, flags agentConfig
 		return nil
 	}
 
+	versionText, isMatching := GetVersion(ctx, cfg)
+	if !isMatching && os.Getenv("TRACETEST_DEV") == "" {
```

Comment:
We could do the exit cli here but I thought it was better to do that from the `cmd` level